third_party/mbedtls/CMakeLists.txt

```diff
@@ -28,14 +28,23 @@
 
 add_library(mbedcrypto)
 
-target_compile_definitions(mbedcrypto PRIVATE
-    ${OT_PRIVATE_DEFINES}
+set(OT_MBEDTLS_CONFIG_FILE "\"mbedtls-config.h\"" CACHE STRING "The mbedTLS config file")
+
+target_compile_definitions(mbedcrypto
+    PUBLIC
+        "MBEDTLS_CONFIG_FILE=${OT_MBEDTLS_CONFIG_FILE}"
+    PRIVATE
+        ${OT_PRIVATE_DEFINES}
 )
 
-target_include_directories(mbedcrypto PRIVATE
-    ${OT_PUBLIC_INCLUDES}
-    ${OT_PRIVATE_INCLUDES}
-    ${PROJECT_SOURCE_DIR}/src/core
+target_include_directories(mbedcrypto
+    PUBLIC
+        ${PROJECT_SOURCE_DIR}/third_party/mbedtls
+        ${PROJECT_SOURCE_DIR}/third_party/mbedtls/repo/include
+    PRIVATE
+        ${OT_PUBLIC_INCLUDES}
+        ${OT_PRIVATE_INCLUDES}
+        ${PROJECT_SOURCE_DIR}/src/core
```

Comment:
The compiling option at here https://github.com/openthread/openthread/pull/4604/files#diff-13df1eb8811b9d6002a25bf0bbddd046R35 is public, it adds the macro `MBEDTLS_CONFIG_FILE` to a target that is depending on `mbedcrypto`. But it should fail because `mbedcrypto`  privately adds `${PROJECT_SOURCE_DIR}/src/core` which includes `openthread-core-config.h` (included by `MBEDTLS_CONFIG_FILE`). It works for now because all targets depending on `mbedcrypto` has already manually added `${PROJECT_SOURCE_DIR}/src/core`.

But, to work with modern CMake, I think this needs to be public.